bpf/tc/attach.go

```diff
@@ -105,6 +105,19 @@ func (ap AttachPoint) AttachProgram() error {
 		return err
 	}
 
+	matchedHash, objHash, err := bpf.CheckAttachedProgs(ap.IfaceName(), ap.FileName())
```

Comment:
Should include the direction (ingress/egress) in the args to `CheckAttachedProgs` since we want it to be precise enough to capture a different value for each attachment point.  Each interface has two programs attached so you'll need two files; one for each direction.

I think you should pass in the fully-qualified file name since this file owns that information and it's not good to duplicate the path in your new file.